### PR TITLE
Set the `SWIFT_VERSION` on resource bundle targets that contain compiled sources and use Swift

### PR DESCRIPTION
**Pull request by dnkoutso**
closes https://github.com/CocoaPods/CocoaPods/issues/7950

### CODE REVIEW

File: lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb [hunk 1]
```diff
@@ -739,8 +747,8 @@ def add_placeholder_target
           #
           def add_swift_static_library_compatibility_header_phase(native_target)
             if custom_module_map
-              raise Informative, 'Using Swift static libraries with custom module maps does not yet work.' \
-                                 "Please build #{pod_target.label} as a framework or remove the custom module map for the time being."
+              raise Informative, 'Using Swift static libraries with custom module maps does not yet work. ' \
+                                 "Please build `#{target.label}` as a framework or remove the custom module map for the time being."
```
Comment (dnkoutso) [inline lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: @segiddins this was actually crashing. `pod_target` is not a declared variable here.

Comment (dnkoutso) [reply lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb]: thankfully a different test caught it.

File: spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb [hunk 2]
```diff
@@ -625,6 +628,12 @@ class PodsProjectGenerator
                     bc.build_settings['TARGETED_DEVICE_FAMILY'].should == '1,2'
                   end
                 end
+
+                it 'sets the Swift version build setting when resources bundle contains sources and target has Swift' do
+                  @bundle_target.build_configurations.each do |bc|
+                    bc.build_settings['SWIFT_VERSION'].should == '3.2'
+                  end
+                end
```
Comment (amorde) [inline spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb]: Should test that is isn't set when there aren't any compile sources, or is that covered by other tests?

Comment (dnkoutso) [reply spec/unit/installer/xcode/pods_project_generator/pod_target_installer_spec.rb]: it is not but can do :D

**Comment by dnkoutso** [comment]
@amorde done

**Pull request merged**